cfme/scripting/bz.py

```diff
@@ -0,0 +1,149 @@
+To then set qe_test_coverage on those BZs, you can do,
+```python,
+miq bz coverage --set <directory>
+```
+"""
+import os
+from collections import namedtuple
+
+import click
+import pytest
+import yaml
+
+from cfme.utils.blockers import BZ
+from cfme.utils.log import logger
+
+
+def get_report(directory):
+    pytest.main([
+        "--use-provider", "complete",
+        "--long-running",
+        "--use-template-cache",
+        "--collect-only",
+        "--generate-bz-report", directory
+    ])
+    # read the generated yaml
+    with open("bz-report.yaml", "r") as stream:
+        info = yaml.load(stream, Loader=yaml.BaseLoader)
+    return info
+
+
+def get_qe_test_coverage(info, open_only=True):
+    """
+    Given info (what is returned from yaml.load on bz-report.yaml),
+    This function screens and returns only BZs which are OPEN, if open_only=True, otherwise it
+    returns all BZs
+    """
+    BZTestCoverage = namedtuple("BZTestCoverage", ["id", "qe_test_coverage"])
+
+    bz_list = []
+    for bug_id in info.keys():
+        if open_only and info[bug_id]["is_open"] == "false":
+            continue
+        # get qe_test_coverage_flag
+        qe_test_coverage = "?"
+        for flag in info[bug_id]["flags"]:
+            if flag["name"] == "qe_test_coverage":
+                qe_test_coverage = flag["status"]
+                break
+        # append the BZ
+        bz_list.append(BZTestCoverage(id=bug_id, qe_test_coverage=qe_test_coverage))
+
+    return bz_list
+
+
+def cleanup():
+    logger.info("Removing the BZ report file, bz-report.yaml")
+    try:
+        os.remove("bz-report.yaml")
+    except OSError:
+        logger.exception("bz-report.yaml not found")
+
+
+@click.group(help="Functions for generating reports on BZs included in test suite metadata")
+def main():
+    pass
+
+
+@main.command(help="Generate BZ report on BZs that have coverage given a directory")
+@click.argument("directory", default="cfme/tests/")
+def report(directory):
+    logger.info("Generating a BZ report in bz-report.yaml")
+    get_report(directory)
+
+
+@main.command(help="List open/closed BZs that have test coverage")
+@click.argument("directory", default="cfme/tests/")
+@click.option("--open", "open_bzs", is_flag=True, help="list open BZs")
+@click.option("--all", "all_bzs", is_flag=True, help="list all BZs with coverage")
+def list(directory, open_bzs, all_bzs):
+    info = get_report(directory)
+
+    if open_bzs:
+        val = "true"
+        text = "are open"
+    elif all_bzs:
+        val = None
+        text = "have coverage"
+    else:
+        val = "false"
+        text = "are closed"
+
+    if not all_bzs:
+        bz_list = [bug_id for bug_id in info.keys() if info[bug_id]["is_open"] == val]
+    else:
+        bz_list = [bug_id for bug_id in info.keys()]
+
+    if bz_list:
+        print("The following BZ's {}: \n{}".format(text, "\n".join(bz_list)))
```

Comment:
Required:
Please use `click.echo()` in state of `print`. Its good practice for cli application writing.